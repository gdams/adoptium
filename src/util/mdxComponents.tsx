import React from 'react';
import { FaCheck, FaTimes } from 'react-icons/fa';

import GuestPost from '../components/GuestPost';
import CodeBox from '../components/CodeBox';
import InlineCode from '../components/CodeBox/InlineCode';

const Small = props => <><br/><small className='text-muted' {...props} /></>;

const Tick = () => <FaCheck style={{ color: 'var(--brand4)', marginRight: '0.5rem' }} />;
const Cross = () => <FaTimes style={{ color: 'var(--brand4)', marginRight: '0.5rem' }} />;

// Wrapper to create a collapsible element using native markdown
const Collapsible = props => {
  return (
    // if props.title is defined add summary
    props.title ? <details className='collapsible pb-3'><summary>{props.title}</summary><div className='content' {...props}></div></details> :
    <details className='collapsible pb-3' {...props} />
  )
}

// Wrapper to add a class name around an element
const Class = props => {
  return (
    <div className={props.name} {...props} />
  )
}
  
export const mdxComponents = {
    Small,
    Tick,
    Cross,
    Collapsible,
    Class,
    GuestPost,
    code: InlineCode,
    InlineCode: InlineCode,
    pre: CodeBox,
    blockquote: props => <blockquote style={{ paddingLeft: '1.5rem', borderLeft: '.3rem solid hsla(0,0%,0%,0.9)' }} className='blockquote' {...props} />,
    table: props => <table className='table table-hover' {...props} />,
    thead: props => <thead className='table-dark' {...props} />,
    li: props => <li style={{ marginBottom: '1.5em' }} {...props} />
};
