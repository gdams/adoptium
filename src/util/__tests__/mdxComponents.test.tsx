import { mdxComponents } from '../mdxComponents';
import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

describe('mdxComponents', () => {
    it('small renders correctly', () => {
        const { container } = render(mdxComponents.Small({ children: 'test' }));
        expect(container).toMatchSnapshot();
    });

    it('tick renders correctly', () => {
        const { container } = render(mdxComponents.Tick());
        expect(container).toMatchSnapshot();
    });

    it('cross renders correctly', () => {
        const { container } = render(mdxComponents.Cross());
        expect(container).toMatchSnapshot();
    });

    it('collapsible renders correctly', () => {
        const { container } = render(mdxComponents.Collapsible({ children: 'test' }));
        expect(container).toMatchSnapshot();
    });

    it('collapsible renders correctly - with title', () => {
        const { container } = render(mdxComponents.Collapsible({ title: 'test', children: 'test' }));
        expect(container).toMatchSnapshot();
    });

    it('class renders correctly', () => {
        const { container } = render(mdxComponents.Class({ name: 'test', children: 'test' }));
        expect(container).toMatchSnapshot();
    });
});
