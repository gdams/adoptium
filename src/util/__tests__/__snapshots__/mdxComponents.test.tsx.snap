// Vitest Snapshot v1

exports[`mdxComponents > class renders correctly 1`] = `
<div>
  <div
    class="test"
    name="test"
  >
    test
  </div>
</div>
`;

exports[`mdxComponents > collapsible renders correctly - with title 1`] = `
<div>
  <details
    class="collapsible pb-3"
  >
    <summary>
      test
    </summary>
    <div
      class="content"
      title="test"
    >
      test
    </div>
  </details>
</div>
`;

exports[`mdxComponents > collapsible renders correctly 1`] = `
<div>
  <details
    class="collapsible pb-3"
  >
    test
  </details>
</div>
`;

exports[`mdxComponents > cross renders correctly 1`] = `
<div>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    style="margin-right: 0.5rem;"
    viewBox="0 0 352 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
    />
  </svg>
</div>
`;

exports[`mdxComponents > small renders correctly 1`] = `
<div>
  <br />
  <small
    class="text-muted"
  >
    test
  </small>
</div>
`;

exports[`mdxComponents > tick renders correctly 1`] = `
<div>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    style="margin-right: 0.5rem;"
    viewBox="0 0 512 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
    />
  </svg>
</div>
`;
